hw2/controllers/GroupController.spec.js

```diff
@@ -0,0 +1,329 @@
+    expect(res.json).toHaveBeenCalledWith({
+      message: "Group with id 1 not found!",
+    });
+  });
+
+  it("should return a 500 status if an error occurs", async () => {
+    const error = new Error("Database error");
+    GroupModel.findByPk.mockRejectedValue(error);
+
+    const req = {
+      params: { id: 1 },
+    };
+    const res = {
+      json: jest.fn(),
+      status: jest.fn(),
+    };
+
+    await getGroupByID(req, res);
+
+    expect(GroupModel.findByPk).toHaveBeenCalledWith(1);
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(res.json).toHaveBeenCalledWith({ message: "Server error" });
+  });
+});
+
+describe("postGroup", () => {
+  it("should create a new group", async () => {
+    const req = {
+      body: { name: "Group 1", permissions: "read" },
+    };
+    const res = {
+      json: jest.fn(),
+      status: jest.fn(),
+    };
+
+    await postGroup(req, res);
+
+    expect(GroupModel.create).toHaveBeenCalledWith(req.body);
+    expect(res.status).toHaveBeenCalledWith(201);
+    expect(res.json).toHaveBeenCalled();
+  });
+
+  it("should return a 500 status if an error occurs", async () => {
+    const error = new Error("Database error");
+    GroupModel.create.mockRejectedValue(error);
+
+    const req = {
+      body: { name: "Group 1", permissions: "read" },
+    };
+    const res = {
+      json: jest.fn(),
+      status: jest.fn(),
+    };
+
+    await postGroup(req, res);
+
+    expect(GroupModel.create).toHaveBeenCalledWith(req.body);
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(res.json).toHaveBeenCalledWith({ message: "Server error" });
+  });
+});
+
+describe("putGroup", () => {
+  it("should update the group with the specified ID", async () => {
+    const req = {
+      params: { id: 1 },
+      body: { name: "Group 1 Updated", permissions: "write" },
+    };
+    const res = {
+      sendStatus: jest.fn(),
+    };
+
+    await putGroup(req, res);
+
+    expect(GroupModel.update).toHaveBeenCalledWith(
+      {
+        name: req.body.name,
+        permissions: req.body.permissions,
+      },
+      {
+        where: {
+          id: req.params.id,
+        },
+        returning: true,
+      }
+    );
+    expect(res.sendStatus).toHaveBeenCalledWith(200);
+  });
+
+  it("should return a 500 status if an error occurs", async () => {
+    const error = new Error("Database error");
+    GroupModel.update.mockRejectedValue(error);
+
+    const req = {
+      params: { id: 1 },
+      body: { name: "Group 1 Updated", permissions: "write" },
+    };
+    const res = {
+      sendStatus: jest.fn(),
+    };
+
+    await putGroup(req, res);
+
+    expect(GroupModel.update).toHaveBeenCalledWith(
+      {
+        name: req.body.name,
+        permissions: req.body.permissions,
+      },
+      {
+        where: {
+          id: req.params.id,
+        },
+        returning: true,
+      }
+    );
+    expect(res.sendStatus).toHaveBeenCalledWith(500);

```

Comment:
better to use constants, 
https://stackoverflow.com/questions/18765869/accessing-http-status-code-constants